File: student-next/src/types/progress.ts
import { LessonSummary } from "./lesson";

export interface ProgressResponse {
  course_id: number | null;
  course_slug?: string | null;
  course_title?: string | null;
  completed_lessons: number;
  total_lessons: number;
  percent: number;
  completed_modules: any[];
  completed_module_names: string[];
  completed_lesson_titles: string[];
  certificates: any[];
  next_lesson: LessonSummary | null;
  progress_map: Record<number, string>;
  xp_total: number;
  xp_today: number;
  streak_days: number;
  goal_today: {
    target?: number;
    completed?: boolean;
    goal_type?: string;
  };
}

File: backend/app/schemas/progress.py
from typing import Any, Optional

from pydantic import BaseModel, Field

from .lesson import LessonSummary


class ProgressPayload(BaseModel):
    course_id: Optional[int]
    course_slug: str | None = None
    course_title: str | None = None
    completed_lessons: int = 0
    total_lessons: int = 0
    percent: int = 0
    completed_modules: list[Any] = Field(default_factory=list)
    completed_module_names: list[str] = Field(default_factory=list)
    completed_lesson_titles: list[str] = Field(default_factory=list)
    certificates: list[Any] = Field(default_factory=list)
    next_lesson: LessonSummary | dict | None = None
    progress_map: dict = Field(default_factory=dict)
    xp_total: int = 0
    xp_today: int = 0
    streak_days: int = 0
    goal_today: dict = Field(default_factory=dict)


__all__ = ["ProgressPayload"]
